Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs

```diff
@@ -63,3 +90,39 @@ updatePerson personId req = do
       mbEncEmail
       (req.deviceToken)
   pure APISuccess.Success
+
+createSosDetails :: (EsqDBFlow m r, EncFlow m r) => Id Person.Person -> SosReq -> m SosRes
+createSosDetails personId req = do
+  sosDetails <- buildSosDetails personId req
+  runTransaction $ QSos.create sosDetails
+  return $
+    SosRes
+      { sosId = sosDetails.id
+      }
+
+updateSosDetails :: (EsqDBFlow m r, EncFlow m r) => Id Person.Person -> SosFeedbackReq -> m APISuccess.APISuccess
+updateSosDetails personId req = do
+  sosDetails <-
+    QSos.findById (req.sosId)
+      >>= fromMaybeM SosIdDoesNotExist
```

Comment:
You can send sosId also while throwing this error

Reply:
> You can send sosId also while throwing this error

Done